### PR TITLE
Zoom interaction

### PR DESCRIPTION
**Pull request by kbusemeyer**
Created a function that allows zooming to a specific given level (as opposed to calculating a delta) and centers it around the given x/y

Also adds a prop to handle a context menu interaction

### CODE REVIEW

File: src/components/graph-view.js [hunk 1]
```diff
@@ -672,6 +659,16 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
...
     }
 
+    if (
+      !d3.event.shiftKey &&
+      d3.event.type === 'contextmenu' &&
+      onContextMenu
+    ) {
+      const xycoords = d3.mouse(d3.event.target);
+
+      onContextMenu(xycoords[0], xycoords[1], d3.event);
```
Comment (PankajTrifacta) [inline src/components/graph-view.js:670]: did we check with the graph controls slider, or we dont have plan to support that. Also do we not need that fit to screen option

File: src/components/graph-view.js [hunk 2]
```diff
@@ -260,8 +261,7 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
       nextProps.edges !== this.props.edges ||
       !nextState.componentUpToDate ||
       nextProps.selected !== this.props.selected ||
-      nextProps.readOnly !== this.props.readOnly ||
-      nextProps.layoutEngine !== this.props.layoutEngine
```
Comment (kseth) [inline src/components/graph-view.js]: why did you remove this layoutEngine check here?

File: src/components/graph-view.js (hunk 1 above)
Comment (kseth) [reply src/components/graph-view.js:670]: We don't need to show the graph controls slider and the zoom to fit option will be part of the context menu.

**Review by kseth — APPROVED** [review]

File: src/components/graph-view.js (hunk 2 above)
Comment (kbusemeyer) [reply src/components/graph-view.js]: Complete accident, I think when I was doing clean up I accidentally deleted too much
